### PR TITLE
feat(frontend): DappsCarousel components

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

This PR includes DappsCarousel components.

https://github.com/user-attachments/assets/3a474efe-4a3b-45c7-895f-02591836f3ed

### CODE REVIEW

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	import {
+		dAppDescriptions,
+		type CarouselSlideOisyDappDescription
+	} from '$lib/types/oisyDappDescription';
+
+	export let styleClass: string | undefined = undefined;
+
+	const dappsCarouselSlides = dAppDescriptions.filter(({ carousel }) =>
+		nonNullish(carousel)
+	) as CarouselSlideOisyDappDescription[];
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/dapps/DappsCarousel.svelte:14]: do you need the casting? are you not taking elements of `dAppDescriptions`, that I think it's `OisyDappDescription` type?

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	import Carousel from '$lib/components/carousel/Carousel.svelte';
+	import DappsCarouselSlide from '$lib/components/dapps/DappsCarouselSlide.svelte';
+	import {
+		dAppDescriptions,
+		type CarouselSlideOisyDappDescription
+	} from '$lib/types/oisyDappDescription';
+
+	export let styleClass: string | undefined = undefined;
+
+	const dappsCarouselSlides = dAppDescriptions.filter(({ carousel }) =>
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/dapps/DappsCarousel.svelte:12]: nitpick:  `dAppDescriptions.filter(nonNullish)`

does it work? eheh

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		nonNullish(carousel)
+	) as CarouselSlideOisyDappDescription[];
+</script>
+
+{#if nonNullish(dappsCarouselSlides)}
+	<Carousel {styleClass}>
+		{#each dappsCarouselSlides as dappsCarouselSlide}
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/dapps/DappsCarousel.svelte:19]: same question about casting here

File: src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	$: ({
+		carousel: { text, callToAction },
+		logo,
+		name
+	} = dappsCarouselSlide);
+</script>
+
+<div class="flex">
+	<div class="mr-4 shrink-0">
+		<Img
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte:18]: can you use `Logo` component here? Or it does not fit the needs?

File: src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+			height="64"
+			width="64"
+			rounded
+			src={logo}
+			alt={replacePlaceholders($i18n.dapps.alt.logo, { $dAppname: name })}
+		/>
+	</div>
+	<div>
+		<div class="mb-1">{text}</div>
+		<button
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte:28]: Can you use the `Button` component here? Or it does not fits the needs?

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:14]: OisyDappDescription has `carousel` as optional. So, after filtering elements by this param, we can cast it to CarouselSlideOisyDappDescription where it's required.

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:12]: The filter only checks `carousel`, `.filter(nonNullish)` would look for the whole description to be non-nullish.

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 3 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:19]: Actually, here it's not casting but `for-each` loop

File: src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte (hunk 4 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte:18]: `Logo` has some additional stuff that does not fit to this case

File: src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte (hunk 5 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte:28]: I just tried it - it comes with too many things and apparently I cannot turn it off via props. So, if it's okay with you, I'd just use `button` here

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import Carousel from '$lib/components/carousel/Carousel.svelte';
+	import DappsCarouselSlide from '$lib/components/dapps/DappsCarouselSlide.svelte';
+	import {
+		dAppDescriptions,
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:6]: yeah, makes sense

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:19]: my bad! you are completly right!

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:12]: ah right! all good! tks!

File: src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte (hunk 5 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarouselSlide.svelte:28]: yes, makes sense

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM tks

File: src/frontend/src/lib/components/dapps/DappsCarousel.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/dapps/DappsCarousel.svelte:14]: Thanks for the explanation!

**Pull request merged**